File: 백준/Silver/10866. 덱/덱.py
from collections import deque

N = int(input())
commands = [list(input().split()) for _ in range(N)]
my_deque = deque()

for command in commands:
    if len(command) == 2:
        X = int(command[1])
        if command[0] == 'push_front':
            my_deque.appendleft(X)
        else:
            my_deque.append(X)
    else:
        if command[0] == 'pop_front':
            if len(my_deque) == 0:
                print(-1)
                continue
            print(my_deque.popleft())
        elif command[0] == 'pop_back':
            if len(my_deque) == 0:
                print(-1)
                continue
            print(my_deque.pop())
        elif command[0] == 'size':
            print(len(my_deque))
        elif command[0] == 'empty':
            if len(my_deque) == 0:
                print(1)
            else:
                print(0)
        elif command[0] == 'front':
            if len(my_deque) == 0:
                print(-1)
                continue
            print(my_deque[0])
        elif command[0] == 'back':
            if len(my_deque) == 0:
                print(-1)
                continue
            print(my_deque[-1])

File: 백준/Silver/10866. 덱/덱.js
const input = require('fs').readFileSync(process.version === 'v18.17.1' ? 'input.txt' : '/dev/stdin')
  .toString().trim().split("\n");


const solution = () => {
  const N = +input[0];
  const stack = [];
  const result = [];
  for (let i = 1; i <= N; i++) {
    const [command, number] = input[i].split(" ").map(v => v.trim());

    switch (command) {
      case "push_front":
        stack.unshift(number);
        break;
      case "push_back":
        stack.push(number);
        break;
      case "pop_front":
        const front = stack.shift();
        result.push(front ?? -1);
        break;
      case "pop_back":
        const end = stack.pop();
        result.push(end ?? -1);
        break;
      case "size":
        result.push(stack.length);
        break;
      case "empty":
        result.push(stack.length > 0 ? 0 : 1);
        break;
      case "front":
        result.push(stack.at(0) ?? -1);
        break;
      case "back":
        result.push(stack.at(-1) ?? -1);
        break;
      default:
        break;
    }
  }

  console.log(result.join("\n"));
};

solution();

